ste/mgr-JobMgr.go

```diff
@@ -715,7 +715,14 @@ func (jm *jobMgr) reportJobPartDoneHandler() {
 				(isCancelling && !haveFinalPart) // If we're cancelling, it's OK to try to exit early; the user already accepted this job cannot be resumed. Outgoing requests will fail anyway, so nothing can properly clean up.
 			if shouldComplete {
 				// Inform StatusManager that all parts are done.
-				close(jm.jstm.xferDone)
+
+				jm.jstm.flagMutex.Lock()
+				if !jm.jstm.isXferDoneClosed { //channel is open
```

Comment:
Did we introduce **isXferDoneClosed** variable just to keep status of xferDone channel as closed or open?
If so we can eliminate this variable and add a function to incorporate the same.

func (jm *jobMgr) xferDoneCLosed() bool {
    select {
    case <-jm.jstm.xferDone:
        return true
    default:
        return false
    }
}

Reply:
I modified the implementation to recover gracefully when a panic while sending on a closed channel occurs. So, there is no longer a need for the variable or function to check if the channel is closed.